services/middleware/src/proto/service.proto

```diff
@@ -0,0 +1,29 @@
+// Copyright 2019 The Exonum Team
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package exonum.service.middleware;
+
+import "runtime.proto";
+
+message CheckedCall {
+  string artifact_name = 1;
+  string artifact_version = 2;
+  exonum.runtime.AnyTx inner = 3;
+}
+
+message Batch {
+  repeated exonum.runtime.AnyTx inner = 1;
```

Comment:
The `Batch` doesn't contain `artifact_name` and `artifact_version` because we know that it will be composed correct ?

Reply:
`Batch` is responsible for batching. If the user wants so, he can wrap the `Batch` into `CheckedCall`, and/or wrap internal calls in the `Batch` in `CheckedCall`s.